src/idpi/operators/regrid.py

```diff
@@ -0,0 +1,184 @@
+
+Resampling: typing.TypeAlias = warp.Resampling
+
+CRS_ALIASES = {
+    "geolatlon": "epsg:4326",
+    "swiss": "epsg:21781",
+    "swiss03": "epsg:21781",
+    "swiss95": "epsg:2056",
+    "boaga-west": "epsg:3003",
+    "boaga-east": "epsg:3004",
+}
+
+
+def _get_crs(geo):
+    if geo["gridType"] != "rotated_ll":
+        raise NotImplementedError("Unsupported grid type")
+
+    lon = geo["longitudeOfSouthernPoleInDegrees"]
+    lat = -1 * geo["latitudeOfSouthernPoleInDegrees"]
+
+    return CRS.from_string(
+        f"+proj=ob_tran +o_proj=longlat +o_lat_p={lat} +lon_0={lon} +datum=WGS84"
+    )
+
+
+def _relimit(longitude: float) -> float:
+    if longitude > 180:
+        return longitude - 360
+    return longitude
+
+
+@dc.dataclass
+class RegularGrid:
+    """Class defining a regular grid.
+
+    Attributes
+    ----------
+    crs : CRS
+        Coordinate reference system.
+    nx : int
+        Number of grid points in the x direction.
+    ny : int
+        Number of grid points in the y direction.
+    xmin : float
+        Coordinate of the first grid point in the x direction.
+    xmax : float
+        Coordinate of the last grid point in the x direction.
+    ymin : float
+        Coordinate of the first grid point in the y direction.
+    ymax : float
+        Coordinate of the last grid point in the y direction.
+
+    """
+
+    crs: CRS
+    nx: int
+    ny: int
+    xmin: float
+    xmax: float
+    ymin: float
+    ymax: float
+
+    @classmethod
+    def from_field(cls, field: xr.DataArray):
+        """Extract grid parameters from grib metadata.
+
+        Parameters
+        ----------
+        field : xarray.DataArray
+            field containing the relevant metadata.
+
+        """
+        geo = field.geography
+        obj = cls(
+            crs=_get_crs(geo),
+            nx=geo["Ni"],
+            ny=geo["Nj"],
+            xmin=_relimit(geo["longitudeOfFirstGridPointInDegrees"]),
+            xmax=_relimit(geo["longitudeOfLastGridPointInDegrees"]),
+            ymin=geo["latitudeOfFirstGridPointInDegrees"],
+            ymax=geo["latitudeOfLastGridPointInDegrees"],
+        )
+        if abs(obj.dx - geo["iDirectionIncrementInDegrees"]) > 1e-5:
+            raise ValueError("Inconsistent grid parameters")
+        if abs(obj.dy - geo["jDirectionIncrementInDegrees"]) > 1e-5:
+            raise ValueError("Inconsistent grid parameters")
+        return obj
+
+    @classmethod
+    def parse_regrid_operator(cls, op: str):
+        """Parse fieldextra out_regrid_target string.
+
+        Parameters
+        ----------
+        op : str
+            fieldextra out_regrid_target definition
+            i.e. crs,xmin,ymin,xmay,ymax,dx,dy.
+
+        """
+        crs_str, *grid_params = op.split(",")
+        crs = CRS.from_string(CRS_ALIASES[crs_str])
+        xmin, ymin, xmax, ymax, dx, dy = map(float, grid_params)
+        nx = (xmax - xmin) / dx + 1
+        ny = (ymax - ymin) / dy + 1
```

Comment:
if xmax == xmin, calling dx would cause a division by zero. But I think it's fair to assume this will never happen, right?